### PR TITLE
added separate tests for WeightColumnMixin

### PR DESCRIPTION
**Pull request by limlam96**
linked to https://github.com/lvgig/tubular/issues/322

added separate tests for this mixin, and made code more polars-y

### CODE REVIEW

File: tubular/mixins.py [hunk 1]
```diff
@@ -138,7 +138,11 @@ class WeightColumnMixin:
         class attribute, indicates whether transformer has been converted to polars/pandas agnostic narwhals framework
     """
 
-    polars_compatible = False
+    polars_compatible = True
+
+    def classname(self) -> str:
```
Comment (davidhopkinson26) [inline tubular/mixins.py:143]: Q: I'm probably missing something but why do we need this here? any class which uses this Mixin will inherit this method from BaseTransformer?

File: tests/mixins/test_WeightColumnMixin.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
+import narwhals as nw
```
Comment (davidhopkinson26) [inline tests/mixins/test_WeightColumnMixin.py:1]: P: thanks for adding this test module!

**Review by davidhopkinson26 — CHANGES_REQUESTED** [review]

File: tubular/mixins.py (hunk 1 above)
Comment (limlam96) [reply tubular/mixins.py:143]: ah yeah you're right it wasn't doing anything, actioned here https://github.com/lvgig/tubular/pull/341/commits/65af6b10e16b820f477d114432c19d3b9b35f206

Comment (davidhopkinson26) [reply tubular/mixins.py:143]: Sorry, looks like it is still there? did you push all changes?

Comment (limlam96) [reply tubular/mixins.py:136]: sorry not sure what happened there, must have been working on a random branch, removed now https://github.com/lvgig/tubular/pull/341/commits/0c6996372f1852bfb310102a9348769bcd63743e

**Review by davidhopkinson26 — DISMISSED** [review]
thanks Liam!

**Comment by limlam96** [comment]
FYI had a conflict so just pushed again :)

**Review by davidhopkinson26 — APPROVED** [review]

**Pull request merged**